src/os_regex/os_regex_str.c

```diff
@@ -18,10 +18,6 @@
 /* Check if a specific string is numeric (like "129544") */
 int OS_StrIsNum(const char *str)
 {
-    if (str == NULL) {
```

Comment:
I left this one in there as a fallback in case something changed somewhere else.

Reply:
no, this change was undesigned, the removal of ``` __attribute__((nonnull))``` from ```OS_StrIsNum``` should be the correct way.
i'll rebase